src/main/java/hudson/plugins/sloccount/SloccountBuildAction.java

```diff
@@ -47,10 +45,18 @@ public String getUrlName() {
     }
     
     @Override  
-    public Collection<? extends Action> getProjectActions() {  
-        return this.projectActions;  
-    }  
+    public Collection<? extends Action> getProjectActions() {
+        return this.projectActions;
+    }
 
+    private void buildProjectActions() {
+        if (this.projectActions != null) {
+            return;
+        }
+
+        this.projectActions = Collections.synchronizedList(new ArrayList<SloccountProjectAction>());
```

Comment:
`CopyOnWriteArrayList` list may offer a better performance for read operations, which are dominant here